sphinx/util/inspect.py

```diff
@@ -158,3 +158,26 @@ def is_builtin_class_method(obj, attr_name):
     if not hasattr(builtins, safe_getattr(cls, '__name__', '')):
         return False
     return getattr(builtins, safe_getattr(cls, '__name__', '')) is cls
+
+
+def dumb_signature(functional_object):
+    src = inspect.getsourcelines(functional_object)[0]
+
+    for i, l in enumerate(src):
+        if 'def' in l:
+            break
+    acc = ''
+    for l in src[i:]:
+        acc += l.strip()
+        if '):' in acc:
```

Comment:
Spaces between the function signature and the terminating colon (eg. ```def foo(bar) :```) are perfectly valid Python, but will lead to this loop never being terminated (or, at least, unexpectedly exhausting the full function body.)